### PR TITLE
V0.4 flags_pretty time_feedback to user

### LINKED ISSUE #161: Feature Suggestion: Fix Parsing to use by instead of /by

### PR DESCRIPTION
**Pull request by kevinLamKB**
Changelog 📜 

- commands now support natural language! e.g. add gardens by tomorrow tag WHEREGOTTIME
-  for now does not support escape character / auto recognizing that a some flag words is actually name. so doing add gardens by the bay by christmas will only use the last flag (by christmas) as the input to parse. 
-  previous flags /by /on /at are no longer recognized, use the same flags as before but without /
-  date parsing still accepts 12/12/17 11:30 type of format , but now it is in MM/DD/YY instead of the previous DD/MM/YY format.
- cleaned up messages to users, just type any command without any arguments to see the standard format

Logic tests will all fail for now. working on it for next PR

Fixes #161 , #129 , #76

### CODE REVIEW

**Comment by kevinLamKB** [comment]
oops, some of the changes are from mattheus's branch when i merged it into mine

**Comment by ChuaPingChan** [comment]
> now it is in MM/DD/YY instead of the previous DD/MM/YY format

Curious why did you make this change?

**Comment by kevinLamKB** [comment]
i didn't, that's how prettyTime does it. but since thats how our dates are being shown to user in the app anyway i decided to leave it as it is

File: src/main/java/seedu/multitasky/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public AddCommand parse(String args) throws ParseException {
                                                  PREFIX_TO, PREFIX_TAG);
         Calendar startDate = null;
         Calendar endDate = null;
+        Prefix datePrefix;
```
Comment (akd13) [inline src/main/java/seedu/multitasky/logic/parser/AddCommandParser.java:45]: Maybe write header comments explaining what these multiple if-else conditions are doing? it's not very clear just reading through and this method is non-trivial

Comment (akd13) [reply src/main/java/seedu/multitasky/logic/parser/AddCommandParser.java:45]: aah never mind, i see it's explained already

File: src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package seedu.multitasky.logic.parser;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import seedu.multitasky.commons.core.Messages;
+import seedu.multitasky.logic.commands.ListCommand;
+import seedu.multitasky.logic.parser.exceptions.ParseException;
+
```
Comment (akd13) [inline src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java:9]: Collation tag?

**Review by akd13 — APPROVED** [review]

**Review by ChuaPingChan — APPROVED** [review]

File: src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java (hunk 2 above)
Comment (kevinLamKB) [reply src/main/java/seedu/multitasky/logic/parser/ListCommandParser.java:9]: ah this one's from mattheus's branch haha don't forget to collate tag urself @mattheuslee

**Review by mattheuslee — APPROVED** [review]

**Comment by mattheuslee** [comment]
@kevinLamKB Are the GUI tests failing as well? Are there any major changes to the input/output format apart from the change from dd/mm/yy to mm/dd/yy?

**Pull request merged**